x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts

```diff
@@ -72,7 +72,8 @@ export class UserProfileAPIClient {
    */
   public bulkGet<D extends UserProfileData>(params: UserProfileBulkGetParams) {
```

Comment:
Wouldn't it be easier to change UserProfileBulkGetParams to accept a simple array of uids? 

Not aware of us using Sets for REST API clients since it's not supported in JSON anyways. 

Is there a particular reason you want to enforce Set?

Reply:
> Is there a particular reason you want to enforce Set?

It's done mostly for consistency with the server-side API that works with `Set`. The idea here is to use `Set` to stress the uniqueness requirement which is one of the primary use cases for `Set`. Historically the arrays were preferred to `Set`'s in such scenarios to have better cross-browser compatibility (:wave: Internet Explorer), but those days are gone.